mongoengine/document.py

```diff
@@ -366,7 +366,7 @@ def save(self, force_insert=False, validate=True, clean=True,
         if write_concern is None:
             write_concern = {}
 
-        doc_id = self.to_mongo(fields=['id'])
+        doc_id = self.to_mongo(fields=[self._meta['id_field']])
```

Comment:
This change has already been taken care of in https://github.com/MongoEngine/mongoengine/pull/2082, right? Can you please rebase on top of the latest master @divomen?